src/Peripherals/SerialPortPeripheral.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "PeripheralInterface.h"
+
+class QString;
+class QSerialPort;
+
+class SerialPortPeripheral : public PeripheralInterface
+{
+public:
+    SerialPortPeripheral(QSerialPort& serialPort);
+    virtual ~SerialPortPeripheral();
+    bool attemptConnection();
+    void setPortName(QString portName);
+    void sendData(const unsigned char* data, int length);
+
+private:
+    QSerialPort& outputDevice_;
```

Comment:
Oh, thought I had to have it as the same name as the inherited interface. Changed.